src/Infiltrator.jl

```diff
@@ -55,6 +55,7 @@ macro infiltrate(cond = true)
   if __module__ === Core.Compiler && !isdefined(Core.Compiler, :Dict)
     # XXX Dict isn't available in Core.Compiler, so make it available now
     Core.eval(Core.Compiler, :(using .Main: Dict))
+    Core.eval(Core.Compiler, :(setindex!(x::Dict, args...) = Main.setindex!(x, args...)))
```

Comment:
How likely is this to break everything, @aviatesk?

Reply:
I'd say it's acceptable, although we generally should avoid this pattern, since the whole purpose of `Core.Compiler` is to separate the evaluation of `Core.Compiler` code from the other general evaluations. But at this moment we need this kind of hack for external packages to work with `Core.Compiler`, and bunch of packages exactly do this stuff. We have a plan to fix this (by re-unifying `Core.Compiler` and `Base`), but I'm fine with this until we get to the refactor.